e2e-tests/src/driver.rs

```diff
@@ -303,12 +303,21 @@ impl Driver {
 
                 merod.run().await?;
 
-                let _ = e.insert(merod);
+                let node_name = merod.name.clone();
+                e.insert(merod);
+
+                let merod_ref = merods.get(&node_name).unwrap();
```

Comment:
`e.insert` already returns the reference to the inserted item, we don't need a separate get step